.gitignore

```diff
@@ -1,4 +1,43 @@
-/vendor/*
+# CakePHP specific files #
+##########################
 /config/app.php
-/tmp/*
+/config/.env # josegonzalez/php-dotenv
 /logs/*
+/tmp/*
+/tmp/bake/*
```

Comment:
I think it is much better if they exist beforehand to avoid any sources of permission problems creating directories. Also they have no side-effects as they are mandatory by the CakePHP framework. When creating this PR I assumed there would be a backslash on adding so much editor support, rather than this, very basic sane default.

I'll fix some things and if there are no other voices I will remove the tmp directory ignores.